### ISSUE TITLE
Can not name relationship field "Type"

### ISSUE THREAD

**Issue by dkenzik** [webhook/webhook-cms#232]
Logging this quickly, and will produce a smaller use-case/demo when I get a chance (hopefully this is the right repo for it).

It seems that if I create a relationship field and use the name "Type", the generator dies when rendering that particular template:

> Fatal error: Cannot set property _type of #<Object> which has only a getter

Renaming that field to anything else that the CMS thinks is valid will work.

**Comment by LtSquigs** [comment]
Yes this is due to us using _type as a private variable and conflicting with the special relationship field stuff, will talk to mike about a workaround for this (like making a field named type translate to something else)

**Comment by dkenzik** [comment]
Thanks! 

Now I don't need to make a small demo of the issue. :-)
